main-support.ts

```diff
@@ -385,6 +385,17 @@ export function extractAllMetadata(
   extractMetaDataCallback(null);
 }
 
+/**
+ * Iterates ffprobe output to find stream with the best resolution (just width, for now)
+ *
+ * @param metadata  the ffProbe metadata object
+ */
+function getBestStream(metadata) {
+  return metadata.streams.reduce(function (a, b) {
+    return a.width > b.width ? a : b;
+  });
+}
```

Comment:
Eh, I realize now this looks a little funny pulled out into its own function, when it's a one-liner for the most part, but it's less dense this way and if the calculation gets more complicated this is better.